### ISSUE TITLE
About node feature and grouping

### ISSUE THREAD

**Issue by JoeyYoung** [gsyang33/Driple#2]
Hi there,

Thanks for your work, I have some questions related to the graph construction:
1. The node features only consider the tensor size and node type. Still, the computation costs of many ops are also related to other hyperparameters, e.g., kernels, channels... why not take them as features? And the paper mentioned fp16/fp32 feature is contained in Xn, but seems not.
2. I'm confused about the grouping method, take vgg16 - conv7 as an example, why is it ensured that mathematical operations in conv7 will be grouped together? How to set the feature of the grouped node?

Hope to hear from you and see the Input/output feature records soon :)

**Comment by gsyang33** [comment]
Sorry for the delayed response.

1. There are a number of additional potential node features, yes. The features of the current paper are the outcome of feature engineering, which took training difficulty and prediction accuracy into account. We were unable to describe all the details in the publication.

2. You made your point well. The semantics of the original graph are not guaranteed by our grouping algorithm currently. In the paper, we gave a simple and straightforward example to comprehend the design. However, please take note that other nodes in Fig. 8b described by "..." are also grouped into G1 in our research; these additional nodes were challenging to explain in detail.

I appreciate your interest.
